internal/provider/cockroach_cluster_data_source.go

```diff
@@ -207,40 +203,7 @@ func (d *clusterDataSource) Read(
 		return
 	}
 
-	cluster.Name = types.StringValue(cockroachCluster.Name)
-	cluster.CloudProvider = types.StringValue(string(cockroachCluster.CloudProvider))
-	cluster.State = types.StringValue(string(cockroachCluster.State))
-	cluster.CockroachVersion = types.StringValue(cockroachCluster.CockroachVersion)
-	cluster.Plan = types.StringValue(string(cockroachCluster.Plan))
-	cluster.OperationStatus = types.StringValue(string(cockroachCluster.OperationStatus))
-	cluster.UpgradeStatus = types.StringValue(string(cockroachCluster.UpgradeStatus))
-	if cockroachCluster.Config.Serverless != nil {
-		cluster.ServerlessConfig = &ServerlessClusterConfig{
-			SpendLimit: types.Int64Value(int64(cockroachCluster.Config.Serverless.GetSpendLimit())),
-			RoutingId:  types.StringValue(cockroachCluster.Config.Serverless.RoutingId),
-		}
-	}
-	if cockroachCluster.Config.Dedicated != nil {
-		cluster.DedicatedConfig = &DedicatedClusterConfig{
-			MachineType:              types.StringValue(cockroachCluster.Config.Dedicated.MachineType),
-			NumVirtualCpus:           types.Int64Value(int64(cockroachCluster.Config.Dedicated.NumVirtualCpus)),
-			StorageGib:               types.Int64Value(int64(cockroachCluster.Config.Dedicated.StorageGib)),
-			MemoryGib:                types.Float64Value(float64(cockroachCluster.Config.Dedicated.MemoryGib)),
-			DiskIops:                 types.Int64Value(int64(cockroachCluster.Config.Dedicated.DiskIops)),
-			PrivateNetworkVisibility: types.BoolValue(cockroachCluster.GetNetworkVisibility() == client.NETWORKVISIBILITYTYPE_PRIVATE),
-		}
-	}
-
-	for _, r := range cockroachCluster.Regions {
-		reg := Region{
-			Name:      types.StringValue(r.Name),
-			SqlDns:    types.StringValue(r.SqlDns),
-			UiDns:     types.StringValue(r.UiDns),
-			NodeCount: types.Int64Value(int64(r.NodeCount)),
-			Primary:   types.BoolValue(r.GetPrimary()),
-		}
-		cluster.Regions = append(cluster.Regions, reg)
-	}
+	loadClusterToTerraformState(cockroachCluster, &cluster, nil)
```

Comment:
nit: maybe a comment stating that passing in `nil` as plan is intentional?

Reply:
Done.